File: app/src/main/java/com/example/diplom/logic/SearchAdapter.java
package com.example.diplom.logic;


import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;
import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;
import com.example.diplom.R;
import com.example.diplom.models.ProductItem;
import java.util.List;

public class SearchAdapter extends RecyclerView.Adapter<SearchAdapter.ViewHolder> {

    private Context context;
    private List<ProductItem> products;
    private OnProductClickListener listener;

    public interface OnProductClickListener {
        void onProductClick(ProductItem product);
    }

    public SearchAdapter(Context context, List<ProductItem> products, OnProductClickListener listener) {
        this.context = context;
        this.products = products;
        this.listener = listener;
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(context).inflate(R.layout.item_search_result, parent, false);
        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
        ProductItem product = products.get(position);
        holder.nameText.setText(product.getName());
        holder.itemView.setOnClickListener(v -> listener.onProductClick(product));
    }

    @Override
    public int getItemCount() {
        return products.size();
    }

    public static class ViewHolder extends RecyclerView.ViewHolder {
        TextView nameText;

        public ViewHolder(@NonNull View itemView) {
            super(itemView);
            nameText = itemView.findViewById(R.id.nameText);
        }
    }

    public void updateList(List<ProductItem> newList) {
        this.products.clear();
        this.products.addAll(newList);
        notifyDataSetChanged();
    }
}

File: app/src/main/java/com/example/diplom/adapters/SearchAdapter.kt
package com.example.diplom.adapters

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.example.diplom.R
import com.example.diplom.models.ProductItem

class SearchAdapter(
    private val context: Context,
    private val products: MutableList<ProductItem>,
    private val listener: OnProductClickListener
) : RecyclerView.Adapter<SearchAdapter.ViewHolder>() {

    interface OnProductClickListener {
        fun onProductClick(product: ProductItem)
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val view = LayoutInflater.from(context).inflate(R.layout.item_search_result, parent, false)
        return ViewHolder(view)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val product = products[position]
        holder.nameText.text = product.name
        holder.itemView.setOnClickListener { listener.onProductClick(product) }
    }

    override fun getItemCount(): Int = products.size

    class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        val nameText: TextView = itemView.findViewById(R.id.nameText)
    }

    fun updateList(newList: List<ProductItem>) {
        products.clear()
        products.addAll(newList)
        notifyDataSetChanged()
    }
}